### PR TITLE
Rewrite the entire dashboard tab in vue

### PR DESCRIPTION
**Pull request by Duddino**
## Abstract
Rewrite the dashboard tab in vue.
Now that a good portion of the codebase is ported to vue, it's possible to use more of the features. The aim is to split UI and logic, in `vue` and `js` files respectively.
`js` functions should not contain any UI code, this includes `createAlert`, as well as accessing the various DOM elements.

To further decouple the logic from the UI, the top-most component should be only one interfacing with the logic, in this PR `dashboard` interfaces with `Wallet`: for example `CreateWallet/VanityGen/AccessWallet` are not in charge of actually creating the wallet, they just emit an event that asks the parent to import it. This also allows us to easily write unit tests.

In the future, when more tabs are refactored, my proposal is to make the vue components own the various logical classes, for example (This may be subject to change once we actually start implementing it):

- `Dashboard` could own a `Wallet` instance,
- `StakingPage` shouldn't own anything, instead `Dashboard` should pass the relevant information through props and events
- `MasternodeTab` should own `Masternode`, but shouldn't access the wallet to get the various UTXOs, instaed `Dashbaord` should pass them using events and props
- `Governance` shouldn't own a `Masternode`... you get the gist.

To clarify for those unfamiliar with vue `events` and `props`:
- Props are passed down the hierarchy, for example `Dashboard` gives `WalletBalance` the balance through props.
- Events are passed up the hierarchy, for example `WalletBalance` emits a `send` event, which is picked by `Dashboard` and is used to open the `TransferMenu`.

## Testing
All of the Dashboard features need to be tested, using the QA checklist:
- Basic wallet 
- Send + Receive
- Ledger
- Localization (On the dashboard tab)

### CODE REVIEW

**Review by panleone — CHANGES_REQUESTED** [review]
I really like the PR,  a huge step in getting rid of global.js and index.html
Before reviewing the code a few bugs that I found during testing:

- The x button to close the `encrypt your wallet` does not work anymore
- In settings the button `change password` appears even if my wallet does not have a password
- Vanity address generation spams a lot of errors after creating the wallet
- When importing a wallet with seed phrase or anything the input is not cleared after pressing the button "Import wallet"
- this testnet xpub xpub6C4D1qkdkJ5zpsQmeyA2nhS58MeWCsZ332bAEd4qmfUueaHkWsUtGeyT6bGkg3SPzxtMPWK9yy5hqeQGDXsRDDwbP5SFVteYyarFr5dUviz does not load balance/utxo and it is a problem of this PR since on current master it works (I think that the problem is that some internal addresses are not loaded in the address map)

**Comment by Duddino** [comment]
>     * The x button to close the `encrypt your wallet` does not work anymore
> 
>     * In settings the button `change password` appears even if my wallet does not have a password
> 
>     * When importing a wallet with seed phrase or anything the input is not cleared after pressing the button "Import wallet"

Fixed, thank you.
>     * Vanity address generation spams a lot of errors after creating the wallet
>     * this testnet xpub xpub6C4D1qkdkJ5zpsQmeyA2nhS58MeWCsZ332bAEd4qmfUueaHkWsUtGeyT6bGkg3SPzxtMPWK9yy5hqeQGDXsRDDwbP5SFVteYyarFr5dUviz does not load balance/utxo and it is a problem of this PR since on current master it works (I think that the problem is that some internal addresses are not loaded in the address map)
Can't reproduce these two. Are you still getting these?

**Review by JSKitty — CHANGES_REQUESTED** [review]
`Activity` title seems to be left-centered now.

image

Once a wallet has fully synced, the Activity list also doesn't refresh itself, you have to hit `Load More` to make it load.

On hitting `Max`, I get a `domCoin` old DOM related error.
image

On hitting the `QR Scanner` button, although expected to see `No Camera` on my PC, some upstream/parent caller is also throwing an `Undefined`.
image

When typing amounts in the `Send` menu, especially editing the `Fiat/Currency` side, the calculations seem extremely buggy, and it's also impossible to backspace-remove the `0` from the currency side, making it difficult to edit the amount to "Send 10 USD" for example.
buggy-send-amounts

**Comment by Duddino** [comment]
All fixed

**Review by panleone — CHANGES_REQUESTED** [review]
Second round of testing:
- Promo codes cannot be created
- Dashboard is not updated when a new transaction is received
- On testnet ( I did not check mainnet ) wallets are always loaded from network: transaction DB is not used
- When loading a single address (i.e. DLabsktzGMnsK5K9uRTMCF6NoYNY6ET4Bb) sync is not working as expected (partial balance is showed, no notification that wallet is being loaded...)
Then by switching from testnet to mainnet back and forth I managed to:

- I cannot send transactions on testnet balance is always too small
- Without having any wallets loaded If i toggle testnet the dashboard is showed, if then I go back to settings and re-switch to mainnet the dashboard is not showed (the screen remains in settings)
- When I open MPW I can see the button "Delete Wallet" even if I don't have any wallets loaded

**Comment by Duddino** [comment]
Thank you
> Then by switching from testnet to mainnet back and forth I managed to:
    I cannot send transactions on testnet balance is always too small
    Without having any wallets loaded If i toggle testnet the dashboard is showed, if then I go back to settings and re-switch to mainnet the dashboard is not showed (the screen remains in settings)
    When I open MPW I can see the button "Delete Wallet" even if I don't have any wallets loaded
If I do the following:
Login with a wallet -> wait for it to sync -> copy the xpub -> delete wallet -> login with xpub
wallet does not first sync

All of these were because some things were not resetting properly on network switch/wallet deletion, they should all be fixed

File: scripts/network.js [hunk 1]
```diff
@@ -282,6 +279,11 @@ export class ExplorerNetwork extends Network {
                 ? 0
                 : nBlockHeights.sort((a, b) => a - b).at(-1);
         this.fullSynced = true;
+        createAlert('success', translation.syncStatusFinished, 12500);
+        getEventEmitter().emit('sync-status-update', 0, 0, true);
+    }
+    reset() {
+        this.fullSynced = false;
```
Comment (panleone) [inline scripts/network.js:286]: shouldn't we reset also `blocks` and `lastBlockSynced`?

**Review by panleone — CHANGES_REQUESTED** [review]
some nits regarding the code

File: scripts/mempool.js [hunk 2]
```diff
@@ -419,6 +420,7 @@ export class Mempool {
         const cNet = getNetwork();
         cNet.fullSynced = true;
         cNet.lastBlockSynced = nBlockHeights.at(-1);
+        getEventEmitter().emit('sync-status-update', 0, 0, true);
```
[the remark this answers is not in the document]
Comment (Duddino) [reply scripts/mempool.js:423]: This is to alert other parts of the code that the sync is done, even if using the txdb

File: scripts/network.js (hunk 1 above)
Comment (Duddino) [reply scripts/network.js:286]: I think they get reset by `walletFullySync` later on, but it's probably a good idea to reset everything

**Review by panleone — APPROVED** [review]
tACK 46ff9ee506e6cf1f47d532e27d6925e56ec2b400
tested basically everything and I like the code

**Review by Liquid369 — APPROVED** [review]
tACK 46ff9ee506e6cf1f47d532e27d6925e56ec2b400
Working great, tried all the QA from our sheet LGTM

**Review by JSKitty — APPROVED** [review]

**Pull request merged**